src/main/java/emu/grasscutter/game/entity/gadget/GadgetObject.java

```diff
@@ -12,9 +15,35 @@ public GadgetObject(EntityGadget gadget) {
 
     @Override
     public boolean onInteract(Player player, GadgetInteractReq req) {
+        if (this.getGadget() == null) return false;
+        if (this.getGadget().getMetaGadget() == null) return false;
+        val interactID = this.getGadget().getMetaGadget().getInteractId();
+        if (interactID == 0) return false;
+        val interactDataEntry = GameData.getGadgetInteractDataMap().values().stream().filter(x -> x.getInteractId() == interactID).toList().get(0);
+
+        //do the thing
```

Comment:
we should probably add a comment to add that we need to check the conditions in the furuter